admin/tests/test_integration.py

```diff
@@ -430,3 +432,111 @@ def test_sdconfig_enable_https_on_source_interface():
     with open(os.path.join(SD_DIR, 'install_files/ansible-base/group_vars/all/site-specific')) as fobj:    # noqa: E501
         data = fobj.read()
     assert HTTPS_OUTPUT == data
+
+
+# The following is the minimal git configuration which can be used to fetch
+# from the SecureDrop Github repository. We want to use this because the
+# developers may have the git setup to fetch from git@github.com: instead
+# of the https, and that requires authentication information.
+GIT_CONFIG = u'''[core]
+        repositoryformatversion = 0
+        filemode = true
+        bare = false
+        logallrefupdates = true
+[remote "origin"]
+        url = https://github.com/freedomofpress/securedrop.git
+        fetch = +refs/heads/*:refs/remotes/origin/*
+'''
+
+ORIGINAL_DIR = os.getcwd()
+NEW_TMP_DIR = ""
+
+
+# This class is to test all the git related operations.
+class TestGitOperations:
+
+    # We will create a new directory and copy the whole git repo
+    # there. This will help us not to destroy the actual work environment.
+    def setup_method(self, method):
+        global NEW_TMP_DIR
+        NEW_TMP_DIR = tempfile.mkdtemp()
```

Comment:
Is there a reason to not use the [tmpdir fixture](https://docs.pytest.org/en/latest/tmpdir.html) instead?